Formula/litani.rb

```diff
@@ -0,0 +1,58 @@
+class Litani < Formula
+  desc "Metabuild system"
+  homepage "https://awslabs.github.io/aws-build-accumulator/"
+  url "https://github.com/ronakfof/aws-build-accumulator/archive/refs/tags/1.19.0.tar.gz"
+  sha256 "7ace2833db4ccf52db21589b4e91d2e5cffae778eecbdfb12998a204032a8225"
+  license "Apache-2.0"
+
+  depends_on "coreutils" => :build
+  depends_on "mandoc" => :build
+  depends_on "scdoc" => :build
+  depends_on "gnuplot"
+  depends_on "ninja"
+
+  resource "Jinja2" do
+    url "https://files.pythonhosted.org/packages/39/11/8076571afd97303dfeb6e466f27187ca4970918d4b36d5326725514d3ed3/Jinja2-3.0.1.tar.gz"
+    sha256 "703f484b47a6af502e743c9122595cc812b0271f661722403114f71a79d0f5a4"
+  end
+
+  resource "MarkupSafe" do
+    url "https://files.pythonhosted.org/packages/bf/10/ff66fea6d1788c458663a84d88787bae15d45daa16f6b3ef33322a51fc7e/MarkupSafe-2.0.1.tar.gz"
+    sha256 "594c67807fb16238b30c44bdf74f36c02cdf22d1c8cda91ef8a0ed8dabf5620a"
+  end
+
+  resource "voluptuous" do
+    url "https://files.pythonhosted.org/packages/c0/2c/ccbeb25364e3e0c5e4522f13d66e2fc639bb4d4ecdf73be0959552cbecb4/voluptuous-0.12.2.tar.gz"
+    sha256 "4db1ac5079db9249820d49c891cb4660a6f8cae350491210abce741fabf56513"
+  end
+
+  resource "graphviz" do
+    url "https://files.pythonhosted.org/packages/f6/14/aa3ec10e4ab1524ba69f4742ef9cfa01fd668d0840afe5d5e6f708a95031/graphviz-0.19.1.zip"
+    sha256 "09ed0cde452d015fe77c4845a210eb642f28d245f5bc250d4b97808cb8f49078"
+  end
+
+  resource "PyYAML" do
+    url "https://files.pythonhosted.org/packages/a0/a4/d63f2d7597e1a4b55aa3b4d6c5b029991d3b824b5bd331af8d4ab1ed687d/PyYAML-5.4.1.tar.gz"
+    sha256 "607774cbba28732bfa802b54baa7484215f530991055bb562efbed5b2f20a45e"
+  end
+
+  def install
+    resources.each do |r|
+      r.stage do
+        system Formula["python@3.9"].opt_bin/"python3", *Language::Python.setup_install_args(prefix)
+      end
+    end
+    bin.install "lib", "doc", "bin", "test", "templates", "litani"
+    system "cd #{bin}/doc && #{Formula["python@3.9"].opt_bin/"python3"} configure"
+    system "cd #{bin}/doc && ninja"
+    man.install Dir["#{bin}/doc/out/man/*.roff"]
+    doc.install Dir["#{bin}/doc/out/html/index.html"]
+  end
+
+  test do
+    system "which", "litani"
+    system Formula["python@3.9"].opt_bin/"python3", "-c", "import jinja2"
+    system Formula["python@3.9"].opt_bin/"python3", "-c", "import voluptuous"
+    system "#{bin}/test/run", "--fast", "--output-dir", buildpath
```

Comment:
`buildpath` doesn't exist anymore. `bin` should not contain directories. We should also aim for a test the simulates user interaction rather than running unit tests.

Reply:
`#{bin}/test/run` contains `end-to-end` tests which verifies complete functionality on the underlying system.

Reply:
We don't want to run full end to end tests though, we just want to run a single command that a user would run to see if that works. End to end tests are for upstream

Reply:
Resolved